storage/src/main/scala/coop/rchain/storage/Storage.scala

```diff
@@ -0,0 +1,261 @@
+                                value: V,
+                                txn: Option[Txn[ByteBuffer]] = None): Unit = {
+
+    if (key.isInstanceOf[Key]) {
+      termTreeKeysSorted.add(key.asInstanceOf[Key])
+      lmdb.put(key.asInstanceOf[Key].toString, value, txn)
+    } else {
+      if (key.isInstanceOf[String]) {
+        try {
+          val k = new Key(key.asInstanceOf[String])
+          termTreeKeysSorted.add(k)
+        } catch {
+          case e: Throwable => {}
+        }
+      }
+      lmdb.put(key, value, txn)
+    }
+  }
+
+  // The getXx[K](key:K) returns an array of Xs.
+  // This set of methods resist being expressed as a
+  // single generic method due to the lack of genericity
+  // of ByteBuffer.
+
+  def getInts[K: Bbable](
+      key: K,
+      txn: Option[Txn[ByteBuffer]] = None): Option[Array[Int]] = {
+    lmdb.getInts(key)
+  }
+  def getStrings[K: Bbable](
+      key: K,
+      txn: Option[Txn[ByteBuffer]] = None): Option[Array[String]] = {
+    lmdb.getStrings(key)
+  }
+  def getFloats[K: Bbable](
+      key: K,
+      txn: Option[Txn[ByteBuffer]] = None): Option[Array[Float]] = {
+    lmdb.getFloats(key, txn)
+  }
+  def getDoubles[K: Bbable](
+      key: K,
+      txn: Option[Txn[ByteBuffer]] = None): Option[Array[Double]] = {
+    lmdb.getDoubles(key, txn)
+  }
+
+  // delete a key and the value associated with it
+  def deleteKey[K: Bbable](key: K,
+                           txn: Option[Txn[ByteBuffer]] = None): Boolean = {
+
+    if (key.isInstanceOf[Key]) {
+      termTreeKeysSorted.remove(key.asInstanceOf[Key])
+      lmdb.deleteKey(key.asInstanceOf[Key].toString, txn)
+    } else {
+      if (key.isInstanceOf[String]) {
+        try {
+          val k = new Key(key.asInstanceOf[String])
+          termTreeKeysSorted.remove(k)
+        } catch {
+          case e: Throwable => {}
+        }
+      }
+      lmdb.deleteKey(key, txn)
+    }
+  }
+
+  // delete a value associated with a key
+  def delete[K: Bbable, V: Bbable](
+      key: K,
+      value: V,
+      txn: Option[Txn[ByteBuffer]] = None): Boolean = {
+    lmdb.delete(key, value, txn)
+  }
+
+  // update a value associated with a key
+  def update[K: Bbable, V: Bbable](
+      key: K,
+      valueToBeReplaced: V,
+      valueReplaceWith: V,
+      txn: Option[Txn[ByteBuffer]] = None): Boolean = {
+    lmdb.update(key, valueToBeReplaced, valueReplaceWith, txn)
+  }
+
+  // return the size limitation imposed on keys by lmdb
+  def getMaxValueKeySize(): Int = {
+    lmdb.getMaxKeySize()
+  }
+
+  // Return the memory in use by the lmdb database, excluding
+  // other memory that supports the lmdb database
+  def memoryInUse(txn: Option[Txn[ByteBuffer]] = None): Long = {
+    lmdb.memoryInUse(txn)
+  }
+
+  def close(): Unit = {
+    lmdb.close()
+  }
+
+  // delete the data.mdb and lock.mdb files
+  def deleteFiles(): Unit = {
+    lmdb.deleteFiles()
+  }
+
+  protected[storage] def displayUniKeys: Unit = {
```

Comment:
Seems like you'd want to dump to a string which can be printed rather than printing directly (or at worst, implement this as `print dumpUniKeys`)

Reply:
I use that method for testing.  I'll change it to return Array[String].